src/Renderer.php

```diff
@@ -405,12 +405,40 @@ public function processStripeTable($stripe)
         $dataset = $stripe['dataset'] ?? [];
         // Ensure each item in the dataset is an array
         $dataset = !empty($dataset) && is_array($dataset) ? $dataset : [[]];
-        foreach ($dataset as $key => $value) {
-            if (!is_array($value)) {
-                $dataset[$key] = [];
+        $colspans = $stripe['colspans'] ?? [];
+
+        $getColspan = function($x, $y) use ($colspans) {
+            foreach ($colspans as $span) {
+                if ($span['x'] === $x && $span['y'] === $y) {
+                    return $span['span'];
+                }
+            }
+            return 1;
+        };
+
+        $tableData = [];
+        foreach ($dataset as $rowIndex => $row) {
+            array_push($tableData, []);
+            if (!is_array($row)) {
+                continue;
+            }
+
+            $spanBuffer = 0;
+            foreach ($row as $columnIndex => $text) {
+                if ($spanBuffer > 0) {
+                    $spanBuffer--;
+                } else {
+                    $colspan = $getColspan($columnIndex, $rowIndex, $colspans);
```

Comment:
Actually, @mcmillenb, been spending a touch more time with this algorithm. Looks like you are passing in `$colspans`, maybe unnecessarily here?

Reply:
And two other observations:

1. The function assigned to variable `$getColspan` "feels" way more JS than PHP to me, just from a coding style/language use perspective. I'd love to get a second take from @merricks and/or @joshwhitwell.
2. After a couple minutes, I was able to ascertain what was going on here, but it might be nice to layer a few more comments in, given the various "exits" from the algorithm (the continue, and the `$spanBuffer--` decrement, as two examples).

That make sense?